### ISSUE TITLE
Return type of some functions is incorrect.

### ISSUE THREAD

**Issue by tomc974** [TypeStrong/typedoc#571]
Hi!

I have a function with a logic similar to the below function

```typescript
function f(x: number) {
  if (x < 5) return true
  if (x > 20) return false
  return undefined
}
```

* The return type inferred by the TypeScript compiler is `boolean | undefined`
* The return type produced by TypeDoc is `undefined | true | false`

No big deal but still annoying...

**Comment by aciccarello** [comment]
@tomc974, the return types are defined by TypeScript.. If you think the type is incorrect you can create an issue on the [TypeScript Project repo](https://github.com/Microsoft/TypeScript). Since it is not something TypeDoc changes I'll close this.

Because I'm curious, why would you want a `true | false` type rather than a `boolean` type? In my mind those are the same with the possible exception of allowing `null`.

**Comment by tomc974** [comment]
@aciccarello It's the other way around. I *do want* the `boolean | undefined` type but TypeDoc generates  documentation with `undefined | true | false`. Try to run it on the provided example ;-)

**Comment by aciccarello** [comment]
@tomc974 Sorry, I misunderstood.

I've confirmed TypeDoc is outputting differently than TypeScript's implicit definition. If the function is explicitly given a `boolean | undefined` return type, that is respected by TypeDoc.

![image](https://user-images.githubusercontent.com/11273838/28723958-ae42cbb2-7375-11e7-8eb9-e37decb75034.png)
![image](https://user-images.githubusercontent.com/11273838/28724002-df758076-7375-11e7-9ca4-049f2902ff21.png)

**Comment by calebsander** [comment]
I see this with an interface method that has the signature `writeValue(buffer: GrowableBuffer, value: VALUE, root?: boolean): void`. The type for `root` shows up instead as `root?: undefined | true | false`. I have an abstract class that implements this interface with an abstract version of the method. Interestingly, it is also broken for the abstract class, but subclasses of the abstract class that implement this method have the correct documentation.

**Comment by Gerrit0** [comment]
Fixed in 0.20.0 beta 30
